### PR TITLE
Use Aleo paths for BFT-related storage (primaries and workers)

### PR DESCRIPTION
**Pull request by niklaslong**
Some of this logic could likely be moved to `aleo-std` at some point.

### CODE REVIEW

File: node/bft-consensus/src/lib.rs [hunk 1]
```diff
@@ -30,17 +30,19 @@ use narwhal_crypto::NetworkKeyPair;
...
 
+use aleo_std::aleo_dir;
 use snarkos_node_consensus::Consensus as AleoConsensus;
 use snarkos_node_messages::{Data, Message, NewBlock};
 use snarkos_node_router::Router;
 use snarkos_node_tcp::protocols::Writing;
 use snarkvm::prelude::{ConsensusStorage, Network};
 
 pub struct BftConsensus<N: Network, C: ConsensusStorage<N>> {
+    // TODO(nkls): remove this
```
Comment (ljedrz) [inline node/bft-consensus/src/lib.rs:45]: are we sure we'll want to remove the numeric primary ID?

File: node/bft-consensus/src/lib.rs [hunk 2]
```diff
@@ -60,8 +62,67 @@ pub enum BftError {
     EyreReport(String),
 }
 
+fn base_path(dev: Option<u16>) -> PathBuf {
+    // Retrieve the starting directory.
+    match dev.is_some() {
+        // In development mode, the ledger is stored in the repository root directory.
```
Comment (ljedrz) [inline node/bft-consensus/src/lib.rs]:
```suggestion
        // In development mode, the ledger is stored in the root directory of the repository.
```

File: node/bft-consensus/src/lib.rs (hunk 1 above)
Comment (niklaslong) [reply node/bft-consensus/src/lib.rs:45]: I _think_ it's only used for dev logs currently, so don't see why not?

File: node/bft-consensus/src/lib.rs [hunk 3]
```diff
@@ -60,8 +62,67 @@ pub enum BftError {
...
+        true => match std::env::current_dir() {
+            Ok(current_dir) => current_dir,
+            _ => PathBuf::from(env!("CARGO_MANIFEST_DIR")),
+        },
+        // In production mode, the ledger is stored in the `~/.aleo/` directory.
```
Comment (ljedrz) [inline node/bft-consensus/src/lib.rs:73]: what about tests that might not directly use the `dev` option? then again, would they use the entire `Validator` :thinking:? just some food for thought

Comment (niklaslong) [reply node/bft-consensus/src/lib.rs:73]: Fair—we might want to revisit this when tackling the CLI structure 🤔

File: node/bft-consensus/src/lib.rs (hunk 1 above)
Comment (ljedrz) [reply node/bft-consensus/src/lib.rs:45]: also the storage paths, which might be useful if we ever need some deep debugging options

Comment (niklaslong) [reply node/bft-consensus/src/lib.rs:45]: Not after this PR 😄 (it's in `dev`).

Comment (ljedrz) [reply node/bft-consensus/src/lib.rs:45]: as long as we keep `dev` in the end :detective:

**Comment by ljedrz** [comment]
I just tested it, and the storage folders are placed in the `consensus` folder; perhaps we could move them to the top-level folder of the `snarkOS` repo? also, I think it would be nice to collect them under a single `.bft-<NETWORK_ID>` folder there, which will come in handy once we have more primaries and/or workers.

**Comment by ljedrz** [comment]
Don't we want the most recent adjustments to apply to non-dev runs as well :detective: ?

**Comment by niklaslong** [comment]
ah misunderstood initially—done 👍

**Review by ljedrz — APPROVED** [review]
LGTM

**Pull request merged**